### PR TITLE
add correlation id to logs in log mediator and during timeouts

### PR DESCRIPTION
**Pull request by fazlan-nazeem**
## Purpose

Fix for wso2/product-ei#3442

### CODE REVIEW

File: modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/passthru/DeliveryAgent.java [hunk 1]
```diff
@@ -243,10 +243,10 @@ public void connected(HttpRoute route, NHttpClientConnection conn) {
     private void tryNextMessage(MessageContext messageContext, HttpRoute route, NHttpClientConnection conn) {
         if (conn != null) {
             try {
-                if (targetConfiguration.isCorrelationLoggingEnabled()) {
+//                if (targetConfiguration.isCorrelationLoggingEnabled()) {
```
Comment (sajithshn) [inline modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/passthru/DeliveryAgent.java]: @fazlan-nazeem : can you please remove commented out lines

File: modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/passthru/TargetErrorHandler.java [hunk 2]
```diff
@@ -135,7 +135,7 @@ public void run() {
                     }
 
                     faultMessageContext.setProperty(PassThroughConstants.NO_ENTITY_BODY, true);
-
+                    faultMessageContext.setProperty(PassThroughConstants.CORRELATION_ID,mc.getProperty(PassThroughConstants.CORRELATION_ID));
```
Comment (indikasampath2000) [inline modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/passthru/TargetErrorHandler.java]: Please fix the code formatting.

File: modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/passthru/ClientWorker.java [hunk 3]
```diff
@@ -182,6 +182,7 @@ public int compare(String o1, String o2) {
                 .getContext().getAttribute(SynapseDebugInfoHolder.SYNAPSE_WIRE_LOG_HOLDER_PROPERTY));
         responseMsgCtx.setProperty(PassThroughConstants.PASS_THROUGH_TARGET_CONNECTION,
                 response.getConnection());
+        responseMsgCtx.setProperty(PassThroughConstants.CORRELATION_ID,outMsgCtx.getProperty(PassThroughConstants.CORRELATION_ID));
```
Comment (indikasampath2000) [inline modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/passthru/ClientWorker.java]: Please fix the code formatting.

File: modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/passthru/DeliveryAgent.java (hunk 1 above)
Comment (fazlan-nazeem) [reply modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/passthru/DeliveryAgent.java]: Fixed

File: modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/passthru/TargetErrorHandler.java (hunk 2 above)
Comment (fazlan-nazeem) [reply modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/passthru/TargetErrorHandler.java]: done

File: modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/passthru/ClientWorker.java (hunk 3 above)
Comment (fazlan-nazeem) [reply modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/passthru/ClientWorker.java]: done

**Comment by bougha1** [comment]
Hi everybody;
Please i need to get property of correlation ID to enrich messages before delivered, but i didn't see any solution for the moment, please can someone help me !? 
Thank u all

**Pull request merged**